src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Heaps.cs

```diff
@@ -57,6 +59,12 @@ sealed partial class MetadataBuilder
             int blobHeapStartOffset = 0,
             int guidHeapStartOffset = 0)
         {
+            // -1 for the 0 we always write at the beginning of the heap:
+            if (userStringHeapStartOffset > UserStringHeapSizeLimit - 1)
```

Comment:
Still off by one? I'm thinking offset/index should not be greater than or *equal* max size (which in this case is further reduced by 1). So should it be `if (userStringHeapStartOffset >= UserStringHeapSizeLimit - 1)`?